pkg/detectors/gitlab/v1/gitlab.go

```diff
@@ -98,33 +110,42 @@ func (s Scanner) verifyGitlab(ctx context.Context, resMatch string) (bool, error
 		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", resMatch))
 		res, err := client.Do(req)
 		if err != nil {
-			return false, err
+			return false, false, err
 		}
 
 		defer res.Body.Close()
 		body, err := io.ReadAll(res.Body)
 		if err != nil {
-			return false, err
+			return false, false, err
 		}
 
 		// 200 means good key and has `read_user` scope
 		// 403 means good key but not the right scope
 		// 401 is bad key
 		switch res.StatusCode {
 		case http.StatusOK:
-			return json.Valid(body), nil
+			return json.Valid(body), false, nil
 		case http.StatusForbidden:
+			// check if the user account is blocked or not
+			var apiResp gitlabMessage
+			if err := json.Unmarshal(body, &apiResp); err == nil {
+				if apiResp.Message == blockedUserMessage {
+					return true, true, nil
+				}
+			}
+
```

Comment:
What if we just find that string in the bodyString ? I would love to see @ahrav's input in this matter.

Reply:
String matching on errors in general makes me uneasy, but if we’re confident in the response structure and error location, it should work. Alternatively, we could use bytes.Contains to check the entire response after consuming the body. I’m not sure about the performance difference, but I suspect bytes.Contains might be slightly faster. JSON decoding in Go is typically slow due to the standard library’s use of reflection. That said, I’d go with whichever option is clearer to everyone.

Reply:
Removed unmarshalling logic